src/librbd/image/ValidatePoolRequest.cc

```diff
@@ -74,7 +74,7 @@ void ValidatePoolRequest<I>::handle_read_rbd_info(int r) {
 
     if (m_out_bl.contents_equal(validated_bl)) {
       // already validated pool
-      finish(0);
+      create_rbd_trash();
```

Comment:
This means that each `rbd create` command grows an additional round-trip to the OSD, because `create_rbd_trash()` is always called, right?  I don't think we want that.

Can `ValidatePoolRequest` continue to finish here and thus avoid the additional round-trip if `trash_add` class method is kept as is?

To make "already validated pool" logic work as before, we need `OVERWRITE RBD INFO` to remain the terminating state.  Could `CREATE RBD TRASH` state be placed between `REMOVE SNAPSHOT` and `OVERWRITE RBD INFO` states to achieve that?

Reply:
> Could `CREATE RBD TRASH` state be placed between `REMOVE SNAPSHOT` and `OVERWRITE RBD INFO` states to achieve that?

Then failure to create the trash object would translate into failure to validate the pool.  This is in contrast to the current approach where the pool becomes validated, but every time a user wants to create a new image, an attempt to create the trash object is made on the side.

Reply:
Agree. Good points!